stac_fastapi/types/stac_fastapi/types/clients/async_core.py

```diff
@@ -0,0 +1,592 @@
+
+    async def get_catalog_children(
+        self, catalog_path: str, **kwargs
+    ) -> stac_types.Children:
+        """Get children by catalog path.
+
+        Called with `GET /catalogs/{catalog_path}/children`.
+
+        Args:
+            catalog_path: Path through hierarchy to catalog.
+
+        Returns:
+            Children.
+        """
+        hierarchy = self.hierarchy_definition.copy()
+        split_path = catalog_path.split("/")
+        selected_catalog = find_catalog(hierarchy, split_path)
+        request: Request = kwargs["request"]
+        base_url = str(request.base_url)
+
+        extension_schemas = [
+            schema.schema_href for schema in self.extensions if schema.schema_href
+        ]
+        catalog_children = [
+            browseable_catalog_page(
+                child,
+                base_url,
+                child["catalog_id"],
+                self.stac_version,
+                self.conformance_classes(),
+                extension_schemas,
+            )
+            for child in selected_catalog["children"]
+            if "catalog_id" in child
+        ]
+        child_collection_ids = [
+            child["collection_id"]
+            for child in selected_catalog["children"]
+            if "collection_id" in child
+        ]
+        all_collections = await self.all_collections(**kwargs)
+        collection_children = [
+            coll
+            for coll in all_collections["collections"]
+            if coll["id"] in child_collection_ids
+        ]
+
+        links = [
+            {
+                "rel": Relations.root.value,
+                "type": MimeTypes.json,
+                "href": base_url,
+            },
+            {
+                "rel": Relations.parent.value,
+                "type": MimeTypes.json,
+                "href": base_url,
+            },
+            {
+                "rel": Relations.self.value,
+                "type": MimeTypes.json,
+                "href": urljoin(base_url, "children"),
+            },
+        ]
+        return stac_types.Children(
+            children=catalog_children + collection_children, links=links
+        )
+
+    async def post_catalog_search(
+        self, search_request: Search, **kwargs
+    ) -> stac_types.ItemCollection:
+        """Catalog refined search (POST).
+
+        Called with `POST /search`.
+
+        Args:
+            search_request: search request parameters.
+
+        Returns:
+            ItemCollection containing items which match the search criteria.
+        """
+        # Pydantic is fine for specifying body parameters but proves difficult to
+        # use for Path parameters. Here, the starlette request is inspected instead
+        request_path = kwargs["request"]["path"]
+        split_path = request_path.split("/")[2:-1]
+        remaining_hierarchy = self.hierarchy_definition.copy()
+        # PGSTAC TODO: Add optional 'catalog' parameter to search, which will determine which
```

Comment:
are any of these PGSTAC TODO laying around the PR important to resolve before we merge?

Reply:
These were initially added to provide some pointers to @bitner about what kinds of functionality could move back into pgstac, but I'm happy to get rid of them!